uproot4/_util.py

```diff
@@ -635,3 +635,56 @@ def awkward_form_of_iter(awkward1, form):
         )
     else:
         raise RuntimeError("unrecognized form: {0}".format(type(form)))
+
+
+def damerau_levenshtein(a, b, ratio=False):
+    """Modified Damerau-Levenshtein distance. Adds a middling penalty
+    for capitalization.
+
+    Parameters
+    ----------
+    a : str
+    b : str
+    ratio : bool, optional, default False
+        Return ratio (len(a)+len(b) - DLH)/(len(a)+len(b))
+
+    Returns
+    -------
+    float
+        Modified Damerau Levenshtein distance
+    """
+    M = [[0] * (len(b) + 1) for i in range(len(a) + 1)]
+
+    for i in range(len(a) + 1):
+        M[i][0] = i
+    for j in range(len(b) + 1):
+        M[0][j] = j
+
+    for i in range(1, len(a) + 1):
+        for j in range(1, len(b) + 1):
+            if a[i - 1] == b[j - 1]:  # Same char
+                cost = 0
+            elif a[i - 1].lower() == b[j - 1].lower():  # Same if lowered
+                cost = 0.5
+            else:  # Different char
+                cost = 2
+            M[i][j] = min(M[i - 1][j] + 1,  # Addition
+                          M[i][j - 1] + 1,  # Removal
+                          M[i - 1][j - 1] + cost  # Substitution
+                          )
```

Comment:
Be sure to run `black` on this. I don't enforce it in the CI because I think automatically changing contributor's code is more confusing than helpful, and the automatically-run `flake8` usually catches formatting issues, at least enough to be a good reminder to run `black`.

Reply:
Ah cool, I didn't realize you switched to black